inference-engine/src/gna_plugin/gna_plugin.cpp

```diff
@@ -487,7 +487,9 @@ void GNAPlugin::UpdateInputScaleFromNetwork(InferenceEngine::CNNNetwork & networ
             auto fp32eq = [](float p1, float p2) -> bool {
                 return (std::abs(p1 - p2) <= 0.00001f * std::min(std::abs(p1), std::abs(p2)));
             };
-            float scaleInput = (fqLayer.getLevels() - 1) / (inputRange.second[0] - inputRange.first[0]);
+            // GNA input is always quantized to int16, so number of levels can't be greater than max uint16
```

Comment:
Where did this limitation come from?

Reply:
I think it comes from GNA plugin implementation to standardize the input precision. Actually it's possible to allow int32 input in GNA plugin implementation but it should be taking into account by the customer during input scale factors calculation in the case of internal scale factors calculation algorithm. It can be quite confusing for the customer since he should understand what is the input precision of the next layer after the input layer (in the most of cases it'll be int16).

Reply:
@elilobanova From a plugin development perspective, we should support different inputs precisions which the user had specified either in network, either using the preprocessing step. That's why I'm asking for a review of 62779/62775 tickets.

I agree that there is no sense to extend legacy quantization using scale factors.

But it is not clear why we are disabling other precisions in the POT scenario. POT should handle different input precisions accordingly taking into account the next op after input. And plugin also should support it.

Reply:
Agree, this limitation looks strange. Can it be solved on POT side? And will it work for int8 precision?

Reply:
@mryzhov we agreed to discuss it offline togather with POT team